common-app-lib/src/cmr/common_app/services/search/params.clj

```diff
@@ -215,6 +216,58 @@
   [concept-type params]
   (default-parse-query-level-params concept-type params))
 
+(defn- generate-conditions
+  "Converts given parameters into query conditions"
+  [context concept-type params options]
+  (map (fn [[param value]]
+           (parameter->condition context concept-type param value options))
+        params))
+
+(defn- create-combined-conditions
+  "When valid spatial parameters are given and the spatial OR option is passed,
+   create proper search conditions"
+  [context concept-type options params spatial-params]
+  (let [spatial-conditions (generate-conditions
+                             context concept-type spatial-params options)
+        combined-spatial-conditions (gc/or-conds spatial-conditions)
+        non-spatial-params (-> params
+                               (data/diff spatial-params)
+                               first)
+        non-spatial-conditions (when (seq non-spatial-params)
+                                 (generate-conditions
+                                  context concept-type non-spatial-params options))
+        combined-non-spatial-conditions (when (seq non-spatial-conditions)
+                                          (gc/and-conds non-spatial-conditions))]
+    (if (and combined-spatial-conditions combined-non-spatial-conditions)
+        (gc/and-conds (conj combined-spatial-conditions combined-non-spatial-conditions))
+        (conj combined-spatial-conditions combined-non-spatial-conditions))))
+
+(defn- params->query-conditions
+  "Appropriately combine query operators based on given options"
+  [context concept-type options params query-attribs]
+  (let [valid-spatial-params [:bounding-box :circle :point :line :polygon :bounding-box]
+        spatial-option (if (= (get-in options [:spatial :or]) "true")
+                           :or
+                           :and)
+        spatial-params (when (seq params)
+                         (select-keys params valid-spatial-params))]
+    (if (and (= :or spatial-option)
+             (seq spatial-params)
+             (or (= :collection concept-type)
```

Comment:
Why do we limit to just collections and granules? We can remove this check and support it for all concept types. Even though only collections and granules have spatial parameters, we don't have to have this check.

Reply:
I was getting errors and added this to fix the issue, but some of the other code I added seems to have covered that case as well so I'll remove it